Automattic/wp-calypso
# Jetpack Manage: Making sure filter selections are cleared or corrected when choosing submenu items

Related to https://github.com/Automattic/jetpack-manage/issues/322

## Proposed Changes

* This PR allows a prop - 'path' - to be passed down into `DashboardOverview` and then into `SitesDashboardV2` via `SitesOverviewContext.Provider`. This path prop contains the correct path whether a filter is selected in the Sites V2 dashboard, or a sites submenu item clicked. This then allows us to check if the path prop has changed, and then either clear the filter or correct the filter if needed.
* In practice, this means that on clicking the 'Favorites' or 'Needs Attention' menu items, the filter will be cleared. If adding a filter on the Favorites page, that filter will work with the selected favorites as it's base. And if clicking on the 'Needs Attention' link, the filter will change to 'Needs Attention'.

## Testing Instructions

* To replicate the original issue, on trunk navigate to `http://jetpack.cloud.localhost:3000/sites`. Select a filter from the filter dropdown. Then, navigate to any of the sub-menu item ('Favorites', 'Needs Attention' or 'All'). Notice that the filter is still showing in the UI (even though not active). 
* To test the fix, apply this PR and follow the same steps. The filter will clear on 'Favorites' and 'All', and 'Needs Attention' will correctly show the 'Needs Attention' filter.

## Pre-merge Checklist
